deblur/workflow.py

```diff
@@ -132,33 +145,41 @@ def remove_artifacts_seqs(seqs_fp,
     verbose: boolean, optional
         If true, output SortMeRNA errors
     """
+    logger = logging.getLogger(__name__)
+    logger.info('remove_artifacts_seqs file %s' % seqs_fp)
+
+    if stat(seqs_fp).st_size == 0:
+        logger.warn('file %s has size 0, continuing' % seqs_fp)
+        return
+
     output_fp = join(working_dir,
                      "%s.no_artifacts" % basename(seqs_fp))
+    blast_output = join(working_dir,
+                        '%s.sortmerna' % basename(seqs_fp))
     aligned_seq_ids = set()
     for i, db in enumerate(ref_fp):
+        logger.debug('running on ref_fp %s working dir %s refdb_fp %s seqs %s'
+                     % (db, working_dir, ref_db_fp[i], seqs_fp))
         # run SortMeRNA
-        app_result = sortmerna_map(
-            seq_path=seqs_fp,
-            output_dir=working_dir,
-            refseqs_fp=db,
-            sortmerna_db=ref_db_fp[i],
-            threads=threads,
-            best=1)
-        # Print SortMeRNA errors
-        stderr_fp = app_result['StdErr'].name
-        if stat(stderr_fp).st_size != 0:
-            if verbose:
-                with open(stderr_fp, 'U') as stderr_f:
-                    for line in stderr_f:
-                        print(line)
-            raise ValueError("Could not run SortMeRNA.")
-
-        for line in app_result['BlastAlignments']:
+        params = ['sortmerna', '--reads', seqs_fp, '--ref', '%s,%s' % (db, ref_db_fp[i])]
+        params.extend(['--aligned', blast_output])
+        params.extend(['--blast', '3', '--best', '1', '--print_all_reads'])
+        params.extend(['-v'])
+
+        with open(blast_output + '.log', "w") as f:
+            res = subprocess.call(params, stdout=f)
+        if not res == 0:
+            logger.critical('sortmerna error on file %s' % seqs_fp)
+            return output_fp
+
+        bfl = open(blast_output+'.blast', 'r')
```

Comment:
use `with open('%s.blast' % bast_output. 'r') as bfl:`
and indent the code below - you can then remove the `bfl.close()` call on line 182